### PR TITLE
neuron: add in-cluster build support and test suite

### PR DESCRIPTION
**Pull request by ybrodsky-rh**
## Summary
- Add `inclusterbuild` test suite (3 tests) for validating KMM in-cluster builds when `DriversImage` is empty
- Update all existing neuron suites (vllm, metrics, upgrade) to use shared `NewDeviceConfigBuilder` helper that branches on `DriversImage` presence
- Update upgrade test to support in-cluster build mode (only `UpgradeTargetVersion` required, no `UpgradeTargetDriversImage`)
- Add `AreAllOperatorsReady` fallback for non-OLM Neuron deployments
- Vendor updated eco-goinfra with `NewBuilderWithInClusterBuild` and optional `DriversImage`
- Fix KMM webhook test to register v1beta1 scheme before direct Module creation

## Test plan
- [x] `inclusterbuild` suite: 3/3 passed (ConfigMap creation, device plugin deployment, Neuron capacity)
- [x] `vllm` suite: 1/1 passed (full inference with TinyLlama on NeuronCores)
- [x] `metrics` suite: 7/7 passed (DaemonSet, ServiceMonitor, Prometheus scraping, hardware info, utilization, memory, per-node)
- [x] `upgrade` suite: 6/6 passed (rolling upgrade from 2.27.4.0 to 2.26.5.0 via in-cluster build)
- [x] All suites tested on ROSA HCP 4.21 with OLM-installed operators (NFD, KMM v2.6.0, Neuron v1.2.0)

🤖 Co-Authored-By: [Claude Code](https://claude.com/claude-code)

## Release Notes

* **New Features**
  * Added support for in-cluster build mode for Neuron drivers as an alternative to pre-built images.
  * New integration test suite for in-cluster build functionality.

* **Tests**
  * Updated existing tests to validate both in-cluster build and pre-built image deployment modes.

### CODE REVIEW

File: tests/hw-accel/neuron/internal/neuronhelpers/deploy.go [hunk 1]
```diff
@@ -153,15 +154,22 @@ func AreAllOperatorsReady(apiClient *clients.Settings, neuronOptions *NeuronInst
...
+			"Neuron operator not ready via OLM check, trying direct deployment check: %v", err)
 
-		return false
+		neuronDeploy, deployErr := deployment.Pull(
+			apiClient, "awslabs-gpu-operator-controller-manager", params.NeuronNamespace)
+		if deployErr != nil || !neuronDeploy.IsReady(OperatorReadinessCheckTimeout) {
+			klog.V(params.NeuronLogLevel).Infof("Neuron operator not ready: %v", deployErr)
+
+			return false
+		}
```
[the remark this answers is not in the document]
Comment (ybrodsky-rh) [reply tests/hw-accel/neuron/internal/neuronhelpers/deploy.go:172]: Fix is not needed here: awslabs-gpu-operator-controller-manager is the correct deployment name used by the Neuron operator (validated during live testing on our own ROSA cluster). aws-neuron-operator is the package/namespace name, not the deployment name.

**Review by ggordaniRed — APPROVED** [review]
